classes-recipe/image_types_qcom.bbclass

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+
+inherit image_types
+
+IMAGE_TYPES += "qcomflash"
+
+QCOM_ESP_IMAGE ?= "esp-qcom-image"
+QCOM_ESP_FILE ?= "${@'efi.bin' if d.getVar('QCOM_ESP_IMAGE') else ''}"
+IMAGE_QCOMFLASH_ESPIMG ?= "${DEPLOY_DIR_IMAGE}/${QCOM_ESP_IMAGE}-${MACHINE}.rootfs.vfat"
+
+# There is currently no upstream-compatible way for the firmware to
+# identify and load the correct DTB from a combined-dtb that contains all
+# dtbs defined in KERNEL_DEVICETREE, so pick the first individual image
+# generated by linux-qcom-dtbbin, unless a different one is specified.
+QCOM_DTB_DEFAULT ?= "${@os.path.basename(d.getVar('KERNEL_DEVICETREE').split()[0][:-4]) if d.getVar('KERNEL_DEVICETREE') else ''}"
+QCOM_DTB_FILE ?= "dtb.bin"
+IMAGE_QCOMFLASH_DTB ?= "${DEPLOY_DIR_IMAGE}/dtb-${QCOM_DTB_DEFAULT}-image.vfat"
+
+QCOM_ROOTFS_FILE ?= "system.img"
+IMAGE_QCOMFLASH_FS_TYPE ??= "ext4"
+IMAGE_QCOMFLASH_ROOTFS ?= "${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${IMAGE_QCOMFLASH_FS_TYPE}"
```

Comment:
I think it would come out easier to read/understand if you do not define IMAGE_QCOMFLASH_ROOTFS, IMAGE_QCOMFLASH_DTB and IMAGE_QCOMFLASH_ESPIMG. Some artifact are in DEPLOY_DIR_IMAGE, some in IMGDEPLOYDIR (rightly), but i find it hard to remember while reading through the code.

Reply:
Sure, removed the vars and used the content directly in the create_qcomflash_pkg function.